tests/js/common/shell/shell-document-babies.js

```diff
@@ -305,30 +306,43 @@ function CollectionDocumentSuiteBabies() {
     /// @brief insert with bad key types
     ////////////////////////////////////////////////////////////////////////////////
 
-    testInsertErrorBadKey: function() {
+    testInsertErrorBadKey: function () {
       var l = [null, false, true, 1, -1, {},
         []
       ];
-      l.forEach(function(k) {
-        var docs = collection.insert([{
+      l.forEach(function (k) {
+        const batch = [{
           _key: "a"
         }, {
           _key: k
-        }]);
-        assertEqual(docs.length, 2);
-        assertEqual(docs[0]._key, "a");
-        assertTrue(docs[1].error);
-        assertEqual(docs[1].errorNum, ERRORS.ERROR_ARANGO_DOCUMENT_KEY_BAD.code);
-        collection.remove("a");
+        }];
+
+        if (!cluster) {
+          const docs = collection.insert(batch);
+          assertEqual(docs.length, 2);
+          assertEqual(docs[0]._key, "a");
+          assertTrue(docs[1].error);
+          assertEqual(docs[1].errorNum, ERRORS.ERROR_ARANGO_DOCUMENT_KEY_BAD.code);
```

Comment:
sequence of parameters should be similar; a third parameter with the full document would be even better.